synapse/models/inet.py

```diff
@@ -268,29 +268,52 @@ def _normPyStr(self, valu):
 
         valu = valu.replace('[.]', '.')
         valu = valu.replace('(.)', '.')
-
-        # strip leading/trailing .
         valu = valu.strip().strip('.')
 
-        if not fqdnre.match(valu):
-            raise s_exc.BadTypeValu(valu=valu, name=self.name,
-                                    mesg=f'FQDN failed to match fqdnre [{fqdnre.pattern}]')
-
-        # Make sure we *don't* get an IP address
+        # Make sure we *don't* get an IPv6 address
         try:
-            socket.inet_pton(socket.AF_INET, valu)
+            v6 = ipaddress.IPv6Address(valu)
             raise s_exc.BadTypeValu(valu=valu, name=self.name,
                                     mesg='FQDN Got an IP address instead')
-        except OSError:
+        except ipaddress.AddressValueError:
             pass
 
+        # strip any invalid leading/trailing characters
+        clean = False
+        fqdn_tt = regex.compile(r'[\w_-]', regex.IGNORECASE | regex.UNICODE)
+
+        while not clean:
```

Comment:
For content which is something like ``😊foo.bar.com`` it now becomes ``foo.bar.com``. This gets messy when you consider that emoji IDNA domains are a thing with a small handful of registrars and we can't currently capture them.